### DISCUSSION TITLE
Automatically maintained symlink directory

### DISCUSSION THREAD

**Discussion by owittek** [pkgxdev/pkgx#600] [Ideas]
Hey!

Currently I see two problems from just trying out tea. First of all not all editors are supported which is understandable in the current development state. The second issue is that most terminal users are likely using some kind of syntax highlighting and auto-completion. Both of these features don't really play well with the current way that tea is handling things.

The solution to the listed problems is quite simple and even already [part of the docs](https://docs.tea.xyz/getting-started/using-tea#other-editors): Symlinking to tea

But it's a tedious manual task to first check if a tool has issues (e.g. neovim with the `mason.nvim` plugin is able to auto-install LSPs and such but uses `node`, `deno`, ... depending on what you're installing) and then manually symlink and maintain it. That's why I'm suggesting to add a `bin` directory to `~/.tea` which contains symlinks to all the tools that have already been installed.

And yes, I'm aware that there is an install-type of solution planned as mentioned here: https://github.com/orgs/teaxyz/discussions/364#discussioncomment-4877018

But I think this could be a quick and simple way to fix this issue ahead of time and deprecate it in case of a better solution being added.

**Comment by mxcl** [comment]
I'll add support for symlinking to ~/.tea/local/bin` today.

* edit: https://github.com/teaxyz/cli/pull/601
* edit: v0.35 deploying: https://github.com/teaxyz/cli/actions/runs/5190247156
* edit: https://github.com/teaxyz/pantry/actions/runs/5190383142
* edit: live

**Reply by mxcl** [reply]
It is our goal to get our magic working across the board. However it is clearly an uphill battle that will take years so this is a sensible inbetween.
